pkg/didcomm/protocol/introduce/service.go

```diff
@@ -415,35 +400,42 @@ func nextState(msg service.DIDCommMsg, rec *record, outbound bool) (state, error
 	}
 }
 
-func (s *Service) currentStateRecord(thID string) (*record, error) {
-	src, err := s.store.Get(thID)
+func (s *Service) currentStateName(ctxID string) (string, error) {
+	src, err := s.store.Get(stateNameKey + ctxID)
 	if errors.Is(err, storage.ErrDataNotFound) {
-		return &record{
-			StateName:       stateNameStart,
-			WaitCount:       initialWaitCount,
-			IntroduceeIndex: introduceeIndexNone,
-		}, nil
+		return stateNameStart, nil
 	}
 
+	return string(src), err
+}
+
+func (s *Service) saveTransited(id string, data Transited) error {
+	src, err := json.Marshal(data)
 	if err != nil {
-		return nil, fmt.Errorf("cannot fetch state from store: thid=%s : %w", thID, err)
+		return fmt.Errorf("marshal transited: %w", err)
 	}
 
-	var r *record
-	if err := json.Unmarshal(src, &r); err != nil {
-		return nil, err
+	return s.store.Put(transitedKey+id, src)
+}
+
+func (s *Service) getTransited(id string) (*Transited, error) {
```

Comment:
I am have trouble groking  the term `transited`

Reply:
Changed it to `TransitionalPayload`

Reply:
As in a payload that will be used between states?

(transitioning between states?)